### PR TITLE
AzureDiskEncryption running windows VM without AAD

### PR DESCRIPTION
**Pull request by SudhakaraReddyEvuri**
### Best Practice Checklist
Check these items before submitting a PR... See the Contribution Guide for the full detail: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md 

1. uri's compatible with all clouds (Stack, China, Government)
1. Staged artifacts use _artifactsLocation & _artifactsLocationSasToken
1. Use resourceGroup().location for resource locations
1. Folder names for artifacts (nestedtemplates, scripts, DSC)
1. Use literal values for apiVersion (no variables)
1. Parameter files (GEN-UNIQUE for value generation and no "changemeplease" values
1. $schema and other uris use https
1. Use uniqueString() whenever possible to generate names for resources.  While this is not required, it's one of the most common failure points in a deployment. 
1. Update the metadata.json with the current date

For details: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/bp-checklist.md

- [x] - Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

### Changelog

*
*
*

### CODE REVIEW

File: 201-encrypt-running-windows-vm-without-aad/azuredeploy.json [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+    },
+    "keyVaultResourceGroup": {
+      "type": "string",
+      "metadata": {
+        "description": "Resource group of the KeyVault"
+      }
+    },
+    "keyEncryptionKeyURL": {
+      "type": "string",
+      "defaultValue": "",
```
Comment (bmoore-msft) [inline 201-encrypt-running-windows-vm-without-aad/azuredeploy.json]: will deployment ever succeed if this defaultValue is used?  If not, just remove it...

File: 201-encrypt-running-windows-vm-without-aad/azuredeploy.json [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+        "description": "Should the OS partition be resized to occupy full OS VHD before splitting system volume"
+      }
+    }
+  },
+  "variables": {
+    "extensionName": "AzureDiskEncryption",
+    "extensionVersion": "2.2",
+    "encryptionOperation": "EnableEncryption",
+    "keyEncryptionAlgorithm": "RSA-OAEP",
+    "keyVaultURL": "[concat('https://', parameters('keyVaultName'), '.vault.azure.net/')]",
```
Comment (bmoore-msft) [inline 201-encrypt-running-windows-vm-without-aad/azuredeploy.json]: use reference to get the uri

File: 201-encrypt-running-windows-vm-without-aad/azuredeploy.json [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+    "keyVaultResourceID": "[concat(subscription().id,'/resourceGroups/',parameters('keyVaultResourceGroup'),'/providers/Microsoft.KeyVault/vaults/', parameters('keyVaultName'))]"
```
Comment (bmoore-msft) [inline 201-encrypt-running-windows-vm-without-aad/azuredeploy.json]: simplify and just use resourceId

resourceId(parameters('keyVaultResourceGroup'), 'Microsoft.KeyVault/vaults/', parameters('keyVaultName'))

File: 201-encrypt-running-windows-vm-without-aad/README.md [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+    <img src="http://azuredeploy.net/deploybutton.png"/>
+</a>
+<a href="https://portal.azure.us/#create/Microsoft.Template/uri/https%3A%2F%2Fraw.githubusercontent.com%2Fazure%2Fazure-quickstart-templates%2Fmaster%2F201-encrypt-running-windows-vm-without-aad%2Fazuredeploy.json" target="_blank">
+    <img src="http://azuredeploy.net/AzureGov.png"/>
+</a>
+<a href="http://armviz.io/#/?load=https%3A%2F%2Fraw.githubusercontent.com%2FAzure%2Fazure-quickstart-templates%2Fmaster%2F201-encrypt-running-windows-vm-without-aad%2Fazuredeploy.json" target="_blank">
+    <img src="http://armviz.io/visualizebutton.png"/>
+</a>
+
+This template enables encryption on a running windows VM without needing AAD application. This template assumes that the VM is located in the same region as the resource group. If not, please edit the template to pass appropriate location for the VM sub-resources.
```
Comment (bmoore-msft) [inline 201-encrypt-running-windows-vm-without-aad/README.md]: doesn't this *have* to be true - i.e. the extension must be deployed in the same location as the VM (resourceGroup doesn't really matter) - if so, then we should provide a parameter for the vmLocation and just default it to [resourceGroup().location]

File: 201-encrypt-running-windows-vm-without-aad/azuredeploy.parameters.json [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+    "keyVaultName": {
+      "value": "GEN-KEYVAULT-NAME"
+    },
+    "keyVaultResourceGroup": {
+      "value": "GEN-KEYVAULT-RESOURCEGROUP-NAME"
+    },
+    "keyEncryptionKeyURL": {
+      "value": "GEN-KEYVAULT-ENCRYPTION-KEY-URI"
+    },
+    "volumeType": {
```
Comment (bmoore-msft) [inline 201-encrypt-running-windows-vm-without-aad/azuredeploy.parameters.json]: remove the non-GEN values and just rely on the defaultValues in the template

File: 201-encrypt-running-windows-vm-without-aad/azuredeploy.json (hunk 1 above)
Comment (SudhakaraReddyEvuri) [reply 201-encrypt-running-windows-vm-without-aad/azuredeploy.json]: Yes, keyEncryptionKeyURL is optional. Hence the defaultValue is empty and deployment will succeed.

File: 201-encrypt-running-windows-vm-without-aad/azuredeploy.json (hunk 3 above)
Comment (SudhakaraReddyEvuri) [reply 201-encrypt-running-windows-vm-without-aad/azuredeploy.json]: Thanks for this suggestion . Fixed it.

File: 201-encrypt-running-windows-vm-without-aad/README.md (hunk 4 above)
Comment (SudhakaraReddyEvuri) [reply 201-encrypt-running-windows-vm-without-aad/README.md]: Removed this statement to avoid confusion.

**Comment by SudhakaraReddyEvuri** [comment]
@bmoore-msft  all your comments are addressed and Travis is clean. Can you please merge the PR?

**Pull request merged**